src/python/pants/engine/legacy/graph.py

```diff
@@ -382,7 +382,8 @@ def __hash__(self):
     return hash(self.address)
 
 
-class TransitiveHydratedTarget(datatype(['root', 'dependencies'])):
+# TODO: are there performance concerns to adding type checks here?
```

Comment:
I assumed part of the reason was because of the lack of support for typed collections, and the comparison to file digests seems appropriate! Will turn this TODO into "add type checks in this file".

Reply:
Not that I know of... :)